core/tests/protractor_utils/editor.js

```diff
@@ -227,6 +228,152 @@ var expectCannotDeleteInteraction = function() {
     '.protractor-test-delete-interaction')).isPresent()).toBeFalsy();
 };
 
+// GADGETS
+
+// Additional arguments may be sent to this function, and they will be
+// passed on to the relevant gadget editor.
+var addGadget = function(panelName, gadgetType, gadgetName) {
+
+  // Bring up the gadget insertion modal for the specified panel.
+  element(
+    by.css('.protractor-test-' + panelName + '-panel-insert-gadget-button'))
+    .click();
+
+  general.waitForSystem();
+
+  // Select the desired gadgetType from the modal.
+  element(
+    by.css('.protractor-test-' + gadgetType + '-gadget-selection-modal'))
+    .click();
+
+  var gadgetNameInput = element(by.model('gadgetDict.gadget_name'));
```

Comment:
If possible use by.css() and add a protractor-specific class name. That way contributors editing the production html can see where the protractor dependencies lie.